src/libraries/System.Private.CoreLib/src/System/Reflection/MethodInvoker.cs

```diff
@@ -2,78 +2,367 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Diagnostics;
+using static System.Reflection.InvokerEmitUtil;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime;
+using static System.Reflection.MethodBase;
+using System.Reflection.Emit;
 
 namespace System.Reflection
 {
-    internal sealed partial class MethodInvoker
+    public sealed partial class MethodInvoker
     {
+        private InvokeFunc_ObjSpanArgs? _invokeFunc_ObjSpanArgs;
+        private InvokeFunc_Obj4Args? _invokeFunc_Obj4Args;
+        private InvokeFunc_RefArgs? _invokeFunc_RefArgs;
+        private InvokerStrategy _strategy;
+        private readonly int _argCount;
+        private readonly RuntimeType[] _argTypes;
+        private readonly InvocationFlags _invocationFlags;
+        private readonly InvokerArgFlags[] _invokerArgFlags;
         private readonly MethodBase _method;
+        private readonly bool _needsByRefStrategy;
+        private readonly bool _isStatic;
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public unsafe object? InlinedInvoke(object? obj, IntPtr* args, BindingFlags invokeAttr)
+        public static MethodInvoker Create(MethodBase method)
```

Comment:
Seems all new public APIs missing docs?

Reply:
Yes I'll add that to the Todos in the PR description. I didn't want to churn on the API in this (large) PR.